src/libraries/Native/Unix/System.Native/pal_random.c

```diff
@@ -108,3 +108,75 @@ void SystemNative_GetNonCryptographicallySecureRandomBytes(uint8_t* buffer, int3
     }
 #endif // HAVE_ARC4RANDOM_BUF
 }
+
+/*
+
+Generate cryptographically strong random bytes.
+
+Return 0 on success, -1 on failure.
+*/
+int32_t SystemNative_GetCryptographicallySecureRandomBytes(uint8_t* buffer, int32_t bufferLength)
+{
+    // Same as GetNonCryptographicallySecureRandomBytes, but don't fall back to a PRNG
+    assert(buffer != NULL);
+
+    static volatile int rand_des = -1;
+    long num = 0;
+    static bool sMissingDevURandom;
+
+    if (!sMissingDevURandom)
+    {
+        if (rand_des == -1)
+        {
+            int fd;
+
+            do
+            {
+#if HAVE_O_CLOEXEC
+                fd = open("/dev/urandom", O_RDONLY | O_CLOEXEC);
+#else
+                fd = open("/dev/urandom", O_RDONLY);
```

Comment:
should read from `/dev/random` since this is the secure version.

Reply:
Except during bootup on distros that don't securely manage random seeds at shutdown, /dev/urandom is plenty secure enough.  Using /dev/random on Linux is a good way to get your process to arbitrarily hang.